neuralhydrology/modelzoo/transformer.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from typing import Dict
+import numpy as np
+import math
+
+import torch
+import torch.nn as nn
+
+from neuralhydrology.modelzoo.head import get_head
+from neuralhydrology.modelzoo.basemodel import BaseModel
+
+LOGGER = logging.getLogger(__name__)
+
+class Transformer(BaseModel):
+    """Transformer model class, which relies on PyTorch's TransformerEncoder class.
+
+    This class implements the encoder of a transformer network with a regression or probabilistic head. 
+
+    The model configuration is specified in the config file using the following options:
+    -- transformer_embedding_dimension : int representing the dimension of the input embedding space. 
+                                         This must be dividible by the number of self-attention heads (transformer_nheads).
+    -- transformer_positional_encoding_type : choices to "add" or "concatenate" positional encoding to other model inputs.
+    -- transformer_positional_dropout: fraction of dropout applied to the positional encoding.
+    -- seq_length : integer number of timesteps to treat in the input sequence.
+    -- transformer_nheads : number of self-attention heads.
+    -- transformer_dim_feedforward : dimension of the feed-fowrard networks between self-attention heads.
+    -- transformer_dropout: dropout in the feedforward networks between self-attention heads.
+    -- transformer_nlayers: number of stacked self-attention + feedforward layers.
+
+    Parameters
+    ----------
+    cfg : Config
+        The run configuration.
+    """
+
+    def __init__(self, cfg: Dict):
+        super(Transformer, self).__init__(cfg=cfg)
+
+        if cfg.embedding_hiddens:
+            LOGGER.warning("## Warning: Embedding settings are ignored. Use EmbCudaLSTM for embeddings")
+
+        input_size = len(cfg.dynamic_inputs + cfg.evolving_attributes + cfg.hydroatlas_attributes +
+                         cfg.static_attributes)
+        if cfg.use_basin_id_encoding:
+            input_size += cfg.number_of_basins
+
+        if cfg.head.lower() == "umal":
+            input_size += 1
+
+        # embedding 
+        self.embedding_dim = cfg.transformer_embedding_dimension
+        self.embedding = nn.Linear(in_features=input_size, 
+                                   out_features=self.embedding_dim)
+
+        # positional encoder
+        self.posistional_encoding_type = cfg.transformer_positional_encoding_type
+        if self.posistional_encoding_type.lower() in ['concat', 'cat', 'concatenate']:
+            self.encoder_dim = self.embedding_dim*2
```

Comment:
`self.encoder_dim` is only used in the `__init__`, so I think there is no need to make it a class attribute.